internal/npm/plugin.go

```diff
@@ -71,6 +71,43 @@ func (p *plugin) Validate() error {
 
 // Exec runs the plugin.
 func (p *plugin) Exec() error {
+	// check for an .npmrc file in the root, if it exists, rename it as it could interfere with our configuration
+	log.Trace("Checking for local .npmrc")
+
+	exists, osErr := p.os.Exists(".npmrc")
+	if osErr != nil {
+		return fmt.Errorf("failed when looking for local .npmrc: %w", osErr)
+	}
+
+	if exists {
+		log.Trace("Renaming local .npmrc")
+
+		osErr := p.os.Fs.Rename(".npmrc", ".tmp-npmrc")
```

Comment:
Not sure which way makes more sense here but does it make sense to rename this file in the local workspace or to rename/move it to something like `/tmp/.npmrc/` and back. If things are manipulated within the workspace and the plugin fails or if this plugin is used in parallel (like in stages) could this risk some weird clobbering issue or something? Not sure if it matters but it's the only concern I had. Failed plugins would possibly leave the npmrc in a weird state.

Reply:
Yeah, I had thought of that, but GoLang doesn't like moving files across devices out of the box. Got the following:
```
FATAL  [0000] failed to create rename local .npmrc: rename .npmrc /tmp/.npmrc: invalid cross-device link
```

Could look into seeing if there's a way to override that.